vsm

```diff
@@ -225,32 +318,68 @@ class State(object):
              data = data.replace(key, value).strip()
 
         data = yaml.safe_load(data)
+
         # Currently we support only lists in yaml at base level
         if issubclass(type(data), list):
             for item in data:
-                self.__parse_items(item)
+                # this empty node serves to group its child(ren) together just
+                # as the list item in the YAML file groups its child(ren)
+                # together
+                block_node = TreeNode(NODE_BLOCK, None)
+                config_tree.add_child(block_node)
+
+                rules = self.__parse_items(item, block_node)
+                if rules:
+                    self.add_rules(rules[0], rules[1])
+
+    def __exec_child_rule(self, rule):
```

Comment:
@treitter Could you explain to me the exact purpose of this method?, I fail to see the effect it has.